### ISSUE TITLE
NLP Training Pipeline with aprender Integration

### ISSUE THREAD

**Issue by noahgift** [paiml/entrenar#8]
## Overview

Add NLP training data pipeline that leverages `aprender` preprocessing capabilities for transformer fine-tuning.

### 1. Text Data Preprocessor
- **Input**: Raw text datasets (JSON, CSV, text files)
- **Processing Pipeline**:
  - Tokenization (aprender::text::tokenize)
  - Stop words filtering (aprender::text::stopwords)
  - Stemming/lemmatization (aprender::text::stem)
  - Text normalization (lowercase, punctuation)
- **Output**: Cleaned text ready for tokenization

### 2. Feature Extraction
- **TF-IDF Vectorization** (aprender::text::vectorize::TfidfVectorizer)
- **Document Embeddings** from LDA topics (aprender::text::topic::LDA)
- **Sentiment Features** (aprender::text::sentiment) as auxiliary labels
- Integration with training batches

### 3. Data Augmentation
- **Synonym Replacement**: Use aprender vocabulary for word substitution
- **Back-translation**: Leverage aprender text pipeline
- **Contextual Augmentation**: Preserve semantic meaning

### 4. Training Integration
- **DataLoader**: Batch processing with aprender preprocessing
- **Multi-task Learning**: 
  - Main task: Next token prediction
  - Auxiliary: Sentiment classification (aprender::text::sentiment)
  - Auxiliary: Topic distribution (aprender::text::topic::LDA)
- **Curriculum Learning**: Start with high-quality aprender-filtered data

## API Design

```rust
use aprender::text::{tokenize, stopwords, vectorize};
use entrenar::data::NLPDataPipeline;

let pipeline = NLPDataPipeline::new()
    .with_tokenizer(WhitespaceTokenizer::new())
    .with_stopwords(StopWordsFilter::english())
    .with_vectorizer(TfidfVectorizer::new())
    .with_sentiment_labels(true)
    .build();

let dataset = pipeline.load_from_json("data/train.json")?;
let batches = dataset.batch(32);
```

## Benefits
- Consistent preprocessing between training (entrenar) and inference (realizar)
- Leverage aprender's battle-tested NLP pipeline (74 tests)
- Multi-task learning improves generalization
- Data augmentation increases training data 3-5x

## Dependencies
- aprender >= 0.7.1 (text module with vectorization, sentiment, topic modeling)
- Integration with existing entrenar training loop

## Related
- aprender #70 (TF-IDF implementation) ✅
- realizar issue for inference pipeline

**Comment by noahgift** [comment]
Implemented via src/hf_pipeline/ with full NLP training pipeline: config, dataset handling, loader (safetensors, teacher models), fine_tune, distillation, export (GGUF, publishing), leaderboard, and trainer. Integrates with aprender through the training config system.
